File: 5. Recursion I/119. Pascal's Triangle II/triangle.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        return self.pascal(rowIndex)
    
    def pascal(self, n):
        if n == 0:
            return [1]
        if n == 1:
            return [1,1]

        row = [None for i in range(n+1)]
        prev = self.pascal(n-1)

        row[0] = 1
        row[n] = 1

        for i in range(1, n):
            row[i] = prev[i-1] + prev[i]

        return row

File: 5. Recursion I/119. Pascal's Triangle II/triangle.cpp
class Solution
{
public:
    vector<int> getRow(int rowIndex)
    {
        return pascal(rowIndex);
    }

    vector<int> pascal(int n)
    {
        if (n == 0)
        {
            return {1};
        }

        if (n == 1)
        {
            return {1, 1};
        }

        vector<int> row(n + 1);
        vector<int> prev = pascal(n - 1);
        row[0] = 1;
        row[n] = 1;

        for (int i = 1; i < n; i++)
        {
            row[i] = prev[i - 1] + prev[i];
        }
        return row;
    }
};